src/screens/settings/SettingsGeneralScreen/SettingsGeneralScreen.tsx

```diff
@@ -0,0 +1,343 @@
+import React from 'react';
+import { ScrollView } from 'react-native';
+
+import { useDispatch, useSelector } from 'react-redux';
+
+import SwitchSetting from '../../../components/Switch/Switch';
+import DisplayModeModal from './modals/DisplayModeModal';
+import GridSizeModal from './modals/GridSizeModal';
+
+import { useSettings } from '../../../hooks/reduxHooks';
+import { useTheme } from '@hooks/useTheme';
+import { setAppSettings } from '../../../redux/settings/settings.actions';
+import { SHOW_LAST_UPDATE_TIME } from '../../../redux/updates/updates.types';
+import DefaultChapterSortModal from '../components/DefaultChapterSortModal';
+import {
+  DisplayModes,
+  displayModesList,
+  LibrarySortOrder,
+} from '@screens/library/constants/constants';
+import { useLibrarySettings } from '@hooks/useSettings';
+import useBoolean from '@hooks/useBoolean';
+import { Appbar, List } from '@components';
+import NovelSortModal from './modals/NovelSortModal';
+import NovelBadgesModal from './modals/NovelBadgesModal';
+import { RootState } from '@redux/store';
+import { NavigationState } from '@react-navigation/native';
+import { getString } from '@strings/translations';
+
+interface GenralSettingsProps {
+  navigation: NavigationState;
+}
+
+const GenralSettings: React.FC<GenralSettingsProps> = ({ navigation }) => {
+  const theme = useTheme();
+  const dispatch = useDispatch();
+
+  const {
+    displayMode = DisplayModes.Comfortable,
+    novelsPerRow = 3,
+    showDownloadBadges = true,
+    showNumberOfNovels = false,
+    showUnreadBadges = true,
+    sortOrder = LibrarySortOrder.DateAdded_DESC,
+  } = useLibrarySettings();
+
+  const transformSortOrder = (text: string) => {
```

Comment:
You can instead create a Object like
```
const sortOrderNameMap = {
  'lastReadAt': 'libraryScreen.bottomSheet.sortOrders.lastRead',
  ...
}
```

and use it like `getString(sortOrderNameMap[sortOrder])`

Reply:
Way better solution, however I used the new Map() function for ts reason